test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -7409,6 +7409,50 @@ BOOST_AUTO_TEST_CASE(builtin_reject_gas)
 	CHECK_ERROR(text, TypeError, "Member \"gas\" not found or not visible after argument-dependent lookup");
 }
 
+BOOST_AUTO_TEST_CASE(gas_left)
+{
+	char const* text = R"(
+		contract C {
+			function f() public returns (uint256 val) { return msg.gas; }
+		}
+	)";
+	CHECK_SUCCESS(text);
+
+	text = R"(
+		contract C {
+			function f() public returns (uint256 val) { return gasleft(); }
+		}
+	)";
+	CHECK_SUCCESS(text);
+
+	text = R"(
+		pragma experimental "v0.5.0";
+		contract C {
+			function f() public returns (uint256 val) { return msg.gas; }
+		}
+	)";
+	CHECK_ERROR(text, TypeError, "Member \"gas\" not found or not visible after argument-dependent lookup in msg");
+}
+
+BOOST_AUTO_TEST_CASE(gasleft_as_identifier)
+{
+	char const* text = R"(
+		contract C {
+			function gasleft() public pure returns (bytes32 val) { return "abc"; }
+			function f() public pure { bytes32 val = gasleft(); assert (val == "abc"); }
```

Comment:
This code is not executed, so the assert is not an actual test.

Reply:
The "assert (val == "abc");" will not trigger an error (I only added it to supress a "unused variable" warning - but if the compiler were to resolve to global "gasleft", which returns an uint256, the assignement "bytes32 val = gasleft();" should trigger an invalid conversion value, so it's still a valid test for the resolution... I will move that part to an EndToEnd test nonetheless.

Reply:
Ah I see - ok!